app/controllers/pieces_controller.rb

```diff
@@ -100,14 +100,12 @@ def game_has_two_players?(game)
   end
 
   def check_test(piece, x, y)
-    return false if piece.can_take?(helpers.get_piece(x, y, piece.game)) && piece.type == 'King'
-    return false if piece.can_take?(helpers.get_piece(x, y, piece.game)) && !piece.puts_self_in_check?(x, y)
-
     if piece.puts_self_in_check?(x, y)
       flash.now[:alert] << 'You cannot put/leave yourself in Check.'
       return false
     end
 
+    return false if piece.can_take?(helpers.get_piece(x, y, piece.game)) && piece.type == 'King'
```

Comment:
Ahh I think I finally figured this out. Because line 109 would return true even if a king is the one putting the opposing king in check and a king can never put a king in check. I still think this is redundant with moving up the if statement above this because a king could never move itself into check to put an opposing king into check but just happy to figure out what it is there for. We can do whatever with this.

Reply:
Yeap, I believe that was my reasoning for it. But it is not needed, so I will remove it and see what the specs think.

Reply:
Removed it and the specs agreed it was ok without it. Simplification!